src/main/java/com/puppycrawl/tools/checkstyle/gui/TreeTable.java

```diff
@@ -143,6 +155,32 @@ private void makeCodeSelection() {
         new CodeSelector(tree.getLastSelectedPathComponent(), editor, linePositionMap).select();
     }
 
+    /**
+     * Generate Xpath.
+     */
+    private void generateXpath() {
+        if (tree.getLastSelectedPathComponent() instanceof DetailAST) {
+            try {
+                final FileText fileText = new FileText(file, StandardCharsets.UTF_8.name());
+                final DetailAST rootAST = (DetailAST) tree.getModel().getRoot();
+                final DetailAST ast = (DetailAST) tree.getLastSelectedPathComponent();
+                final int defaultTabWidth = 4;
+
+                final XpathQueryGenerator queryGenerator = new XpathQueryGenerator(rootAST,
+                    ast.getLineNo(), ast.getColumnNo(),
+                    fileText, defaultTabWidth);
+
+                xpathEditor.setText(queryGenerator.generateXpathQuery(ast));
+            }
+            catch (IOException ex) {
```

Comment:
please catch all exceptions here, we should print problem if any exception happen.

Reply:
Done